src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] object;
+    private int size = 0;
+
+    public ArrayList() {
+        object = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == object.length) {
+            increaseListSize();
+        }
+        object[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index exceeds size of the list");
+        }
+        if (size == object.length) {
+            increaseListSize();
+        }
+        System.arraycopy(object, index, object, index + 1, size - index);
+        object[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
```

Comment:
You do not follow the DRY principle, which tells us "Don't Repeat Yourself". (https://dzone.com/articles/software-design-principles-dry-and-kiss) The logic of your code repeats. You should move it to the separate private method.